version: '3'
name: linguacafe

networks:
    linguacafe:
        driver: bridge

services:
    webserver:
        container_name: linguacafe-webserver
        # https://docs.docker.com/compose/compose-file/build/#illustrative-example
        image: ghcr.io/simjanos-dev/linguacafe-webserver:main
        build:
            context: .
            dockerfile: ./docker/PhpDockerfile
        depends_on:
            mysql:
                condition: service_healthy
        volumes:
            - /your/linguacafe/dict/folder:/var/www/html/storage/app/dictionaries
            - /your/linguacafe/logs/folder:/var/www/html/storage/logs
        ports:
            - "9191:80"
            - "3000:3000"
        networks:
            - linguacafe
        extra_hosts:
            - "host.docker.internal:host-gateway"
    mysql:
        image: mysql:8.0
        container_name: linguacafe-database
        restart: unless-stopped
        tty: true
        ports:
          - "3308:3306"
        volumes:
          - /your/linguacafe/database/folder:/var/lib/mysql
        environment:
          MYSQL_DATABASE: linguacafe
          MYSQL_USER: linguacafe
          MYSQL_PASSWORD: linguacafe
          MYSQL_ROOT_PASSWORD: linguacafe
          SERVICE_NAME: linguacafe
        healthcheck:
            test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
            interval: 5s
            timeout: 5s
            retries: 3
        networks:
            - linguacafe
    python:
        container_name: linguacafe-python-service
        command: /app/manage.py runserver 0.0.0.0:8678
        restart: unless-stopped
        tty: true
        # https://docs.docker.com/compose/compose-file/build/#illustrative-example
        image: ghcr.io/simjanos-dev/linguacafe-python-service:main
        build:
            dockerfile: ./docker/PythonDockerfile
        volumes:
            - /your/linguacafe/dict/folder:/var/www/html/storage/app/dictionaries
        networks:
            - linguacafe
